### PR TITLE
BCA

### PR DESCRIPTION
**Pull request by mdgrs**
Moved to the end of file because I needed the sum function. 

Not much going on: very similar to the bias correction PR. 

I preferred to create a whole new function to better fit with the boostrap function in fitdistr, but I am open to suggestions for improvements.

### CODE REVIEW

File: src/fastmath/stats.clj [hunk 1]
```diff
@@ -1053,3 +1031,66 @@
...
+  {:metadoc/categories #{:extent}}
+  ([vs] (percentile-bca-extent vs 2.5))
+  ([vs ^double p] (percentile-bca-extent vs p (- 100 p)))
+  ([vs p1 p2] (percentile-bca-extent vs p1 p2 :legacy))
+  ([vs ^double p1 ^double p2 estimation-strategy]
+   (let [avs (m/seq->double-array vs)
+         accel (estimate-acceleration avs)
+         m (mean avs)
+        ;;  icdf of the number of bootstrap samples <= the mean
+         ^double z0 (r/icdf r/default-normal (/ (double  (count (filter  #(<= ^double % m) vs))) (count vs)))
```
Comment (genmeblog) [inline src/fastmath/stats.clj:1065]: This (and below) is the same in both function - should be a separate function

File: src/fastmath/stats.clj [hunk 2]
```diff
@@ -1053,3 +1031,66 @@
       :cis (map (fn [^double r]
                   (* ci (m/sqrt (dec (+ r r))))) (reductions (fn [^double acc ^double s]
                                                                (+ acc (* s s))) acf-data))})))
+
+
+(defn- estimate-acceleration
+"Estimates acceleration for BCA bootstrap confidence interval computation"
+([avs]  (let [m (mean avs)
```
Comment (genmeblog) [inline src/fastmath/stats.clj:1038]: `mean` again calculated twice

File: src/fastmath/stats.clj [hunk 3]
```diff
@@ -1053,3 +1031,66 @@
...
+              influence (map #(- ^double % m) avs)
+              num (sum (map m/cb influence))
+              denom (* 6 (m/sqrt (m/cb (sum (map m/sq influence)))))]
+          (/ (double num) denom))))
+
+(defn- accelerated-quantile [^double z0 ^double z1 ^double a]
+  (let [num (+ z0 z1)
+       denom (- 1 (* a num))]
+       (+ z0 (/ (double num) denom))
```
Comment (genmeblog) [inline src/fastmath/stats.clj:1047]: `double` is not necessary here, `num` is floating point number.

**Comment by genmeblog** [comment]
Thanks! Though, I would refactor it further (reducing all duplicates). If you prefer I can make it after merge.

**Comment by genmeblog** [comment]
@mdgrs I merged the code but have one concern about `estimate-acceleration`. In the paper it's a skewness of data but why in your function moments are not scaled to the number of samples. Shouldn't it be something like that?

```clojure
(defn- estimate-acceleration
  "Estimates acceleration for BCA bootstrap confidence interval computation"
  ^double [avs ^double m]
  (let [influence (map (fn [^double v] (- v m)) avs)
        num       (/ (sum (map m/cb influence)) (count avs)) ;; divided by `n`
        denom     (* 6.0 (m/sqrt (m/cb (/ (sum (map m/sq influence)) (count avs)))))] ;; also divided by `n`
    (/ num denom)))
```

**Comment by genmeblog** [comment]
(deleted previous comment, looks I was mistaken)

**Comment by mdgrs** [comment]
> Thanks! Though, I would refactor it further (reducing all duplicates). If you prefer I can make it after merge.

Thanks for merging and sorry I have not been very responsive. If you havent done the further refactoring already, I'll find some time do it this weekend :)

**Comment by genmeblog** [comment]
No problem! I've refactored it slightly already and verified the result with R (coxed::bca). Thanks!

**Pull request merged**